src/function/onlyWhen/onlyWhen.test.ts

```diff
@@ -0,0 +1,20 @@
+import * as assert from 'assert'
+
+import { id } from '..'
+import { onlyWhen } from './onlyWhen'
+
+describe(`onlyWhen`, () => {
+  describe(`given a predicate that is false`, () => {
+    it(`returns default value`, () => {
+      const expected = 0
+      assert.strictEqual(expected, onlyWhen((_: number) => false, expected, id, 1))
```

Comment:
The order of arguments to `strictEqual()` is first `expected`, then `actual`, see https://nodejs.org/api/assert.html#assert_assert_strictequal_actual_expected_message.

Here they are in reverse order.